### ISSUE TITLE
[Bug]: POS Cart View Custom Price hides item

### ISSUE THREAD

**Issue by dstrukt** [btcpayserver/btcpayserver#5218]
### What is your BTCPay version?

1.11.1

### How did you deploy BTCPay Server?

Docker, Locally & https://master.btcpay.host/

### What happened?

Setting a custom price on an amount causes the item to hide in the shop view, and is no longer visible or searchable.

### How did you encounter this bug?

1. Create Shop
2. View Shop, to confirm all items appear
3. Change item price to `Custom`
4. Reload shop, to show items are no longer visible.

### Relevant log output

```shell
N/A
```

### What browser do you use?

Chrome Version 115.0.5790.114 (Official Build) (x86_64)

### Additional information

Will try to upload the video

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by ndeet** [comment]
This was fixed by Kukks in 1.11.1 but the master.btcpay.host was still on the latest rc release. I just updated it to 1.11.1. Can you please confirm it works now? Thank you :green_heart:

**Comment by dennisreimann** [comment]
Custom price items were previously removed on purpose by @Kukks – does it even make sense to bring them back with the cart?

**Comment by dstrukt** [comment]
Tend to agree here @dennisreimann, it doesn't make a lot of sense - as an admin has many other ways of requests a custom payment and you'd never want to expose that in a brick and mortar or eCommerce store - but this was always something we couldn't be certain how people used the POS, so I didn't push to remove, would love if we could.

**Comment by dstrukt** [comment]
@ndeet Thank you for upgrading, that video was on my local build, which as you've mentioned for 1.11.1 does seem to have addressed, but now I'm faced with a new formatting error:

<img width="873" alt="Screen Shot 2023-08-01 at 7 34 48 PM" src="https://github.com/btcpayserver/btcpayserver/assets/6250771/e728f857-44aa-4a34-a667-eea743777ddd">

**Comment by dennisreimann** [comment]
@pavlenex @Kukks The custom amount items got removed in 43096033173c0bd05c03f64047f96bcb8769dda0 and then recently reintroduced in #5204, which now leads to #5212.

As said, I don't think it makes sense to even have them in the cart view, so imho we should now re-remove it again and maybe display a warning on the admin side, where we explain that those items won't show up with the cart view.

Do we have consense here?

**Comment by Kukks** [comment]
> @pavlenex @Kukks The custom amount items got removed in [4309603](https://github.com/btcpayserver/btcpayserver/commit/43096033173c0bd05c03f64047f96bcb8769dda0) and then recently reintroduced in #5204, which now leads to #5212.
> 
> As said, I don't think it makes sense to even have them in the cart view, so imho we should now re-remove it again and maybe display a warning on the admin side, where we explain that those items won't show up with the cart view.
> 
> Do we have consense here?

I agree it makes no sense in its current form, I just did it quick without thought to get the release out.

We can either:
* do as you said
* in the subtotal list item, show an input form allowing the user to specify custom amount there

**Comment by dennisreimann** [comment]
> in the subtotal list item, show an input form allowing the user to specify custom amount there

Let's not overcomplicate it: What if someone adds  more than one of a custom amount item or the POS has multiple custom amount items … this will become tricky very fast and I don't even see a use case for those items in the cart view.

**Comment by dennisreimann** [comment]
@pavlenex Ok, how do we proceed then? As said, I think this complicates it quite a bit, because we'll have to also solve th case in which there are multiple custom amount items then.

**Comment by dennisreimann** [comment]
I found a solution to this and #5212: We would need to let the user/merchant input the price before adding the item to the cart — same would make sense for the minimum amount items. We can then handle all cases correctly and even add multiple items with different custom/monimum prices, like this ….


![Peek 2023-08-04 08-16](https://github.com/btcpayserver/btcpayserver/assets/886/bbf7064c-38a5-403b-baee-01ea6b8adbd0)

**Comment by pavlenex** [comment]
Concept ACK to the solution @dennisreimann

**Comment by ndeet** [comment]
Agree it is better to make it work than to hide it and show a notice on backend even if we do not yet see a use case ourself. The one mentioned for Paralelni makes sense and abusing the tip field for that is not so nice too. :+1:
